server/routes/createGoal/createGoalsController.ts

```diff
@@ -1,188 +1,134 @@
 import createError from 'http-errors'
-import type { Request, RequestHandler, Response } from 'express'
-import { startOfToday } from 'date-fns'
-import type { CreateGoalsForm } from 'forms'
-import logger from '../../../logger'
+import type { RequestHandler } from 'express'
 import CreateGoalsView from './createGoalsView'
-import futureGoalTargetDateCalculator from '../futureGoalTargetDateCalculator'
-import validateCreateGoalsForm from './createGoalsFormValidator'
 import toCreateGoalDtos from '../../data/mappers/createGoalDtoMapper'
 import EducationAndWorkPlanService from '../../services/educationAndWorkPlanService'
+import {
+  CreateGoalFormAction,
+  CreateGoalsForm,
+  GoalCompleteDateOptions,
+  PartialCreateGoalsForm,
+} from './validators/GoalForm'
 
 export default class CreateGoalsController {
   constructor(private readonly educationAndWorkPlanService: EducationAndWorkPlanService) {}
 
-  getCreateGoalsView: RequestHandler = async (req, res, next): Promise<void> => {
-    const { prisonNumber } = req.params
+  getCreateGoalsView: RequestHandler = async (req, res): Promise<void> => {
     const { prisonerSummary } = req.session
 
-    const createGoalsForm = req.session.createGoalsForm || {
-      prisonNumber,
-      goals: [emptyGoal()],
-    }
-    req.session.createGoalsForm = undefined
+    const view = new CreateGoalsView(prisonerSummary, GoalCompleteDateOptions)
```

Comment:
I've changed the date options to an enum. This means that the content for these options is now in the Nunjucks view and content cannot be provided by the controller. The advantage is that the value can now be explicitly validated against by `class-validator`.